source/indexes.txt

```diff
@@ -24,12 +24,167 @@ Optimize Queries by Using Indexes
    
    /indexes/single-field-index
    /indexes/compound-index
+   /indexes/atlas-search-index
 
 Overview
 --------
 
 On this page, you can see copyable code examples that show how to manage
 different types of indexes by using the {+driver-long+}.
 
-.. TODO
+To use an example from this page, copy the code example into the sample 
+application or your own application. Be sure to replace all placeholders in the 
+code examples, such as ``<connection string URI>``, with the relevant values for 
+your MongoDB deployment.
 
+Sample Application
+~~~~~~~~~~~~~~~~~~
+
+You can use the following sample application to test the code on this page. To
+use the sample application, perform the following steps:
+
+1. Ensure you have the {+driver-short+} installed in your project. See the 
+   :ref:`scala-quick-start-download-and-install` guide to learn more.
+
+#. Copy the following code and paste it into a new ``.scala`` file.
+
+#. Copy a code example from this page and paste it on the specified lines in the
+   file.
+
+.. literalinclude:: /includes/indexes/index-starter-code.scala
+   :language: scala
+   :emphasize-lines: 18-21
+
+Single Field Index
+------------------
+
+The following example creates an ascending index on the specified field:
+
+.. literalinclude:: /includes/indexes/index-code-examples.scala
+   :language: scala
+   :start-after: start-index-single
+   :end-before: end-index-single
+
+To learn more about single field indexes, see the :ref:`scala-single-field-index`
+guide.
+
+Compound Index
+--------------
+
+The following example creates a compound index on the two specified fields.
+
+.. literalinclude:: /includes/indexes/index-code-examples.scala
+   :language: scala
+   :start-after: start-index-compound
+   :end-before: end-index-compound
+
+To learn more about compound indexes, see the :ref:`scala-compound-index` guide.
+
+Multikey Index
+--------------
+
+The following example creates a multikey index on the specified array-valued field:
+
+.. literalinclude:: /includes/indexes/index-code-examples.scala
+   :language: scala
+   :start-after: start-create-multikey-index
+   :end-before: end-create-multikey-index
+
+To learn more about multikey indexes, see the :ref:`scala-multikey-index` guide.
```

Comment:
will work once [DOCSP-42340](https://github.com/mongodb/docs-scala/pull/76) is merged